### PR TITLE
Feat client registration front end back end

### PR DESCRIPTION
**Pull request by ehsanansarii**
The Implementation of client registration(frontEnd and backEnd) and client registration for employee I did them in the same branch because they are related to each other.for client registration fe I used the same code of createClient for employee to avoid the code duplication, using auth context.
=>link of the associated tasks:
1-Implement new client registration for employee - front end:
https://polito-se2-21-09.myjetbrains.com/youtrack/agiles/121-5/current?issue=S202109SPG-18
2-Implement client registration - back end:
https://polito-se2-21-09.myjetbrains.com/youtrack/agiles/121-5/current?issue=S202109SPG-26
3-Implement client registration - front end:
https://polito-se2-21-09.myjetbrains.com/youtrack/agiles/121-5/current?issue=S202109SPG-27

### CODE REVIEW

File: server/app.js [hunk 1]
```diff
@@ -106,6 +109,13 @@ app.post(
   clientHandlers.createClientHandler
 );
 
+app.post(
+  buildAPIPath("/clients/signup"),
+  clientHandlers.signupClientHandlerValidatorChain,
```
Comment (G-Salviati) [inline server/app.js]: it should be called signupClientValidatorChain

File: server/dao/client.js [hunk 2]
```diff
@@ -68,3 +70,24 @@ exports.createClient = (db, fullName, phoneNumber, email, address, wallet) => {
...
+exports.signupClient = async (
+  db,
+  fullName,
+  phoneNumber,
+  email,
+  password,
+  address,
+  wallet
+) => {
+  const hashedPwd = await bcrypt.hash(password, saltRounds);
```
Comment (G-Salviati) [inline server/dao/client.js]: probably is better if you hash password not here but before. in the handler. Not sure. You can search the best practice

**Review by G-Salviati — CHANGES_REQUESTED** [review]

File: server/dao/client.js (hunk 2 above)
Comment (ehsanansarii) [reply server/dao/client.js:83]: @G-Salviati Yes, you are right.but I tried to do it before, it took like 3 hours but I couldn't solve it, then I did this.

**Review by G-Salviati — APPROVED** [review]

**Review by G-Salviati — COMMENTED** [review]
there are still conflicts

**Pull request merged**